hoomd/hpmc/ShapeMoves.h

```diff
@@ -146,18 +146,30 @@ template<typename Shape> class PythonShapeMove : public ShapeMoveBase<Shape>
         for (unsigned int i = 0; i < m_params[type_id].size(); i++)
             {
             Scalar stepsize = this->m_step_size[type_id];
-            Scalar a = fmax(-stepsize, -(m_params[type_id][i]));
-            Scalar b = fmin(stepsize, (1.0 - m_params[type_id][i]));
+            Scalar a = -stepsize;
+            Scalar b = stepsize;
+
             hoomd::UniformDistribution<Scalar> uniform(a, b);
             Scalar r = hoomd::detail::generate_canonical<double>(rng);
             Scalar x = (r < this->m_move_probability) ? uniform(rng) : 0.0;
-            m_params[type_id][i] += x;
+            // Reflect trial moves about boundaries
+            if (m_params[type_id][i] + x > 1)
+                {
+                m_params[type_id][i] = 2 - (m_params[type_id][i] + x);
+                }
+            else if (m_params[type_id][i] + x < 0)
+                {
+                m_params[type_id][i] = x - m_params[type_id][i];
```

Comment:
Jen and I discussed offline and we think we need this math to reflect across 0:

```suggestion
                m_params[type_id][i] = - (m_params[type_id][i] + x);
```

Reply:
This is correct - the initial code made the assumption x > 0, which is not true for proposed moves that would bring m_params[type_id] < 0